annotation-solr/src/main/java/eu/europeana/annotation/solr/service/impl/SolrAnnotationServiceImpl.java

```diff
@@ -584,13 +593,21 @@ private SolrQuery solrUniquenessQuerySemanticTagging(Annotation anno, boolean no
 	
 	private SolrQuery solrUniquenessQuerySimpleTagging(Annotation anno, boolean noSelfDupplicate) {
       SolrQuery query = new SolrQuery();
-      query.setQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getValue() + "\"");
-      query.addFilterQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.TAGGING.getOaType() + "\"");
+      query.setQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.TAGGING.getOaType() + "\"");
+      if(anno.getTarget()!=null) {
+        List<String> targetValues = extractUriValues(anno.getTarget());
+        if(targetValues!=null) {
+          for(String target: targetValues) {
+            query.addFilterQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + target + "\"");
```

Comment:
we should probably switch to use the target_record_uri, still need to agree this with Hugo

Reply:
see comments for solrUniquenessQuerySemanticTagging

Reply:
Improved.